coordinator/src/node/unrealized_pnl.rs

```diff
@@ -0,0 +1,68 @@
+use crate::db;
+use crate::node::Node;
+use crate::position::models::Position;
+use anyhow::Context;
+use anyhow::Result;
+use diesel::r2d2::ConnectionManager;
+use diesel::r2d2::PooledConnection;
+use diesel::PgConnection;
+use rust_decimal::Decimal;
+use time::OffsetDateTime;
+use trade::bitmex_client::BitmexClient;
+use trade::bitmex_client::Quote;
+use trade::cfd::calculate_pnl;
+use trade::Direction;
+
+pub async fn sync(node: Node) -> Result<()> {
+    let mut conn = node.pool.get()?;
+
+    let positions = db::positions::Position::get_all_open_or_closing_positions(&mut conn)?;
+    let current_quote = BitmexClient::get_quote(&OffsetDateTime::now_utc())
+        .await
+        .context("Failed to fetch quote from BitMEX")?;
+
+    for position in positions.iter() {
+        if let Err(e) = sync_position(&mut conn, position, current_quote.clone()) {
+            tracing::error!(position_id=%position.id, ?current_quote, "Failed to update position's unrealized pnl in database: {e:#}")
+        }
+    }
+
+    Ok(())
+}
+
+fn sync_position(
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+    position: &Position,
+    quote: Quote,
+) -> Result<()> {
+    let current_price = match position.direction {
+        trade::Direction::Long => quote.bid_price,
+        trade::Direction::Short => quote.ask_price,
+    };
```

Comment:
I can't reason about this. I assume the `direction` corresponds to the trader's direction. Given that this is the coordinator's perspective, does this affect which price it is. I would welcome a comment here.

Reply:
Since it is the price that the trader will get, we have to use that price. We cannot argue from the side of the coordinator here, otherwise we would try to close the position with two different prices.
I'll see to extract a function that we can use throughout the code - it's not the first time this code triggers a lengthy thinking process :)